pkg/cloudprovider/aws/apis/v1alpha1/provider.go

```diff
@@ -74,6 +74,11 @@ type LaunchTemplate struct {
 	// BlockDeviceMappings to be applied to provisioned nodes.
 	// +optionals
 	BlockDeviceMappings []*BlockDeviceMapping `json:"blockDeviceMappings,omitempty"`
+	// base64 encoded string that will be applied to provisioned nodes as UserData.
+	// UserData must be in the appropriate format based on the specified AMIFamily. Karpenter will merge certain fields into
+	// this UserData to ensure nodes are being provisioned with the correct configuration.
+	// +optional
+	UserData *string `json:"userData,omitempty"`
```

Comment:
Nit: Comment should start with the name of the element it describes for automatic doc generation to work best:

```golang
// UserData is a base64 encoded string that will be ....
```
See https://go.dev/blog/godoc